workflows/differentialabundance.nf

```diff
@@ -73,6 +73,8 @@ if (run_gene_set_analysis) {
     } else if (params.gprofiler2_run) {
         if (!params.gprofiler2_token && !params.gprofiler2_organism) {
             error("To run gprofiler2, please provide a run token, GMT file or organism!")
+        } else {
+            ch_gene_sets = [[]]     // If one of token or organism is provided, make gene_sets empty (nested because of .first() in the gprofiler call)
```

Comment:
```suggestion
            ch_gene_sets = Channel.of([])
```

... for consistency with:

```
 ch_gene_sets = Channel.of(gene_sets_files).map { file(it, checkIfExists: true) }
```

Also, the logic for this block would be simpler if we just did this straight after:

```
if (run_gene_set_analysis) {
```

Then we could remove the last `else`.

Reply:
Ah of course!